lib/local_daily_digest.js

```diff
@@ -0,0 +1,436 @@
+								cIn_nameFamily:[user.nameFamily],
+								cIn_nameGiven: [user.nameGiven],
+								cIn_organization: [user.organization],
+								cIn_bool: true
+							});
+						}
+					});
+
+					checkedOutUsers.forEach(function(user){
+						if(country === user.userProfile.locationId)	
+							checkOutTemp.push({
+								cOut_location: user.userProfile.location,
+								cOut_id: [user.userProfile._id],
+								cOut_profile: [user.globalProfile],
+								cOut_locationId: user.userProfile.locationId,
+								cOut_nameFamily:[ user.userProfile.nameFamily],
+								cOut_nameGiven: [user.userProfile.nameGiven],
+								cOut_organization: [user.userProfile.organization],
+								cOut_bool: true
+							});
+					});
+				});
+
+				//Group the checked in and checked out users based on manager locations
+				finalOutput.push({
+					managerID: managerProfile,
+					checkedInUsers: checkInTemp,
+					checkedOutUsers: checkOutTemp
+				});
+
+			});
+			return cb();
+		},
+		function (cb){
+			sortMailer(finalOutput);
+		}		
+	]);
+}	
+
+function sortMailer(finalOutput){
+
+	var tempProfileTuple=[];
+	async.series([
+		//for each manager, get a list of the users checked in and checked out of the required locations and send them to the mailer
+		function(cb){
+			if(finalOutput.length != 0){
+				finalOutput.forEach(function(managerEditor){
+					if((managerEditor.checkedInUsers.length != 0 )|| (managerEditor.checkedOutUsers.length != 0)){
+						tempProfileTuple.push(managerEditor);
+					}
+				});
+			}		
+			else
+				return cb();
+			return cb();
+		},
+		function(cb){
+			tempProfileTuple.forEach(function(tuple){
+				mailTemplate(tuple);
+			});
+		}
+	]);
+}
+
+
+function mailTemplate(tuple){
+
+	var managerEmail  = '',
+		location=[],
+		currentCheckedInUser = [];
+		currentCheckedOutUser=[],
+		mailOptions ={},
+		sortedCheckIn =[],
+		sortedCheckOut = [],
+		target =[],
+		tempSortedCheckIn = [],
+		final = [],
+		org_out = [],
+		org_in =[],
+		tempSortedCheckOut = [];
+
+	managerEmail = tuple.managerID.userid.replace(/(_\d+)$/,'');
+	async.series([	
+		function (cb){
+
+			// group the checked in and checked out users by location
+			//Hence sending one email perlocation for the checkedin and checkedout users
+			tuple.managerID.dailyDigest.forEach(function(location){
+				tuple.checkedInUsers.forEach(function(user){
+					if(user.cIn_locationId === location){
+						currentCheckedInUser.push(user);
+					}
+				});
+
+				tuple.checkedOutUsers.forEach(function(user){
+					if(user.cOut_locationId === location){
+						currentCheckedOutUser.push(user);
+					}
+				});
+			});
+			return cb();
+		},
+		function(cb){
+
+			//edge case scenario, a user checks in, checks out and checks in again in less then 24 hours
+			//Hence has duplicate entries. Remove the duplicates
+			sortedCheckIn = _.filter(currentCheckedInUser, function (element, index) {
+			    for(index += 1; index < currentCheckedInUser.length; index += 1) {
```

Comment:
This bit basically removes any duplicates that may exist because a user checks in multiple times over the course of 24 hours